src/patterns/structural/small-crates.md

```diff
@@ -26,10 +26,11 @@ fine-grained dependencies.
   has both versions 1.0 and 0.5. Since the `Url` from `url:1.0` and the `Url`
   from `url:0.5` are different types, an HTTP client that uses `url:0.5` would
   not accept `Url` values from a web scraper that uses `url:1.0`.
+  Use [workspace.dependencies](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-dependencies-table) to neutralize.
```

Comment:
This only helps in your crates and it's just a developer experience improvement, not in crates download from crates io. Right? 

Also it's not clear what "neutralize" means. 
Probably "helps in enforcing the same crate version across all your workspace members" or similar is more clear

Reply:
so not sure, may be need to rework whole section, because.
1. there is no problem of hell with one workspace and different compilation roots(if foo and bar package have different compilation trees)
2.  workspace deps allow to mitigate aligning version if possible
3. cargo patch allows to mitigate some compilation issues for versions.
4. even one compilation root can depend on 2 version of same lib, as long as there is no unified call stack for both.

So Rust is efficient in solving dependency hell, way more better of others.

![image](https://github.com/user-attachments/assets/1032c6f6-3e02-4977-a9b8-17ac49a2cdd8)


![image](https://github.com/user-attachments/assets/69adc0dd-d28f-4441-ba15-2c3026acfb7b)

Reply:
fixed, please take a look